packages/protocol/contracts/governance/Governance.sol

```diff
@@ -648,6 +650,29 @@ contract Governance is
   }
   /* solhint-enable code-complexity */
 
+  /**
+   * @notice Revoke votes on all proposal of sender in the referendum stage.
+   * @return Whether or not all votes of an account was successfully revoked.
+   */
+  function revokeVotes() external nonReentrant returns (bool) {
+    address account = getAccounts().voteSignerToAccount(msg.sender);
+    Voter storage voter = voters[account];
+    for (uint256 i = 0; i < dequeued.length; i = i.add(1)) {
+      VoteRecord storage voteRecord = voter.referendumVotes[i];
+      (Proposals.Proposal storage proposal, Proposals.Stage stage) = requireDequeuedAndDeleteExpired(
+        voteRecord.proposalId,
+        i
+      );
+      require(stage == Proposals.Stage.Referendum, "Incorrect proposal state");
+      proposal.updateVote(voteRecord.weight, 0, voteRecord.value, Proposals.VoteValue.None);
+      proposal.networkWeight = proposal.networkWeight - voteRecord.weight;
```

Comment:
@yorhodes How to reduce the networkWeight? I am not sure how to calculate the networWeight of a proposal.

Reply:
the `networkWeight` isn't actually reduced until the voter unlocks
this should be updated as it is elsewhere
```suggestion
proposal.networkWeight = getLockedGold().getTotalLockedGold();
```